pkg/kv/kvserver/batcheval/cmd_delete_range.go

```diff
@@ -56,6 +56,12 @@ func declareKeysDeleteRange(
 		latchSpans.AddNonMVCC(spanset.SpanReadOnly, roachpb.Span{
 			Key: keys.RangeDescriptorKey(rs.GetStartKey()),
 		})
+
+		// Obtain a read only lock on range key stats update key to serialize with
+		// range tombstone GC requests.
+		latchSpans.AddNonMVCC(spanset.SpanReadOnly, roachpb.Span{
+			Key: keys.MakeRangeIDPrefixBuf(rs.GetRangeID()).RangeTombstoneStatsUpdateKey(),
```

Comment:
We only need to set this when `UseRangeTombstone` is set.

We'll need this for all other operations that can affect range keys too, i.e. `ClearRange`, `RevertRange`, and `AddSSTable`. The last one is a bit unfortunate -- we could maybe look at the given SST stats, but I'm not sure we want to rely on them for correctness.

Reply:
> We only need to set this when UseRangeTombstone is set.

That should be fine, it's already in the if with other 'UseRangeTombstone' related latches.